Data model for Clazz work

varunasingh opened the pull request:
Just the models

mikedawson reviewed (commented):
My comment is re. the comment entities. I think we should have one comment entity. Comments are not specific to a submission. All private comments (e.g. those that are to/from the teacher) should be shown on the assignment. That isn't separate.

I think the entity should just be 'Comment'. I think we could (and probably should) reuse the comments in the future on other items (e.g. comments on content, discussion boards, etc).

varunasingh commented:
https://youtu.be/WkCsG-0CD6Y

varunasingh committed:
commit before merge

varunasingh committed:
Merge branch 'dev-crud-nav' into dev-crud-classwork

varunasingh committed:
Quiz and submit persisting OK.

varunasingh committed:
Presenter Tests

varunasingh committed:
commit before merge

varunasingh committed:
Merge branch 'dev-crud-nav' into dev-crud-classwork

varunasingh committed:
Fixes on fragment and updated Jvm and espresso outline

varunasingh committed:
With Espresso testing

varunasingh committed:
Update tests

varunasingh committed:
Commit before merge

varunasingh committed:
Merge branch 'dev-crud-nav' into dev-crud-classwork

varunasingh committed:
Removing Assignment. Replacing terminologies

varunasingh committed:
Refactor

varunasingh committed:
Passing tests OK

varunasingh committed:
Presenter test re test

varunasingh committed:
Pointing to primary server

varunasingh committed:
Merge branch 'dev-crud-nav' into dev-crud-classwork

varunasingh committed:
Espresso fix

varunasingh committed:
Commit before merge

varunasingh committed:
Merge branch 'dev-crud-nav' into dev-crud-classwork

varunasingh committed:
Student ClazzWork progress WIP

varunasingh committed:
Tests List progress

varunasingh committed:
Tests complete for Progress List

varunasingh committed:
Merge branch 'dev-crud-nav' into dev-crud-classwork

varunasingh committed:
Merge branch 'dev-crud-nav' into dev-crud-classwork

varunasingh committed:
Commit before merge

varunasingh committed:
Merge branch 'dev-crud-nav' into dev-crud-classwork

varunasingh committed:
Making things build after merge

varunasingh committed:
Testing and fixes . WIP

mikedawson committed:
Work-in-progress implementation of AccountManager

varunasingh committed:
View modes and more work on the whole fragment

mikedawson committed:
Merge branch 'dev-crud-nav' into dev-crud-nav-login

mikedawson committed:
Add account manager tests

varunasingh committed:
Return and next feature OK

varunasingh committed:
Return and marked flow fix

mikedawson committed:
Work in progress fixing onboarding presenter

mikedawson committed:
Merge branch 'dev-crud-nav' into dev-crud-nav-splashscreen

Commit:
fixed selection of language in onboardingScreen

Commit:
converted to viewpager2

Commit:
activity test for onboarding

Commit:
rtl supported natively now

Commit:
updated test to check locale

Commit:
updated presenter test

mikedawson committed:
Bump timeout on ContentEntryEdit2FragmentTest when importing a zip

Commit:
Merge branch 'primary'

mikedawson committed:
Remove rules from multidex-config.pro so vanillamultidex can be used for testing

Commit:
updated logo, added partnership to onboarding and fixed systemLocale selection

Commit:
added the logo to splash screen

mikedawson committed:
Remove duplicate strings

Commit:
splash screen white background

mikedawson committed:
Remove unused strings. Add Varuna's unused string finder to tools dir.

mikedawson committed:
Add class work strings

mikedawson committed:
Merge branch 'primary' into dev-crud-nav-splashscreen

mikedawson committed:
Update OnBoarding and SplashScreen layouts. Add Tajik to supported languages list.

mikedawson committed:
Minor update to Onboarding screen to increase space available in landscape view

mikedawson committed:
Merge pull request #150 from UstadMobile/dev-crud-nav-splashscreen

Initial merge to fix critical issues with the previous version

varunasingh committed:
Tests for Marking working in debug mode

mikedawson committed:
Merge branch 'primary' into dev-crud-nav-login

kileha3 committed:
Theming: Minor change to make rebranding a lot easier

kileha3 committed:
Theming: Change from using android color to using app's color

kileha3 committed:
Theming: minor change on element color to reference it's color from color resource:

mikedawson committed:
Merge pull request #151 from UstadMobile/primary-theme-enhancement

Theme enhancement

mikedawson committed:
Login WiP commit

mikedawson committed:
Adjust EpubContentPresenter to avoid throwing exceptions when a valid table of contents is not present (e.g. old Epub2 style)

mikedawson committed:
Fix context arguments on VideoContentFragment

varunasingh committed:
Merge branch 'dev-crud-nav' into dev-crud-classwork
and Marking espresso test complete

kileha3 committed:
Theming: Minor upadate which makes dark version more flexible

kileha3 committed:
Merge branch 'primary' of https://github.com/UstadMobile/UstadMobile into primary-theme-enhancement

kileha3 committed:
Ttheming: Minor update -WIP

varunasingh committed:
ProgressList espresso tests complete

kileha3 committed:
Theming: Finalizing theming task with minor change on dark mode template

mikedawson committed:
Merge pull request #152 from UstadMobile/primary-theme-enhancement

Primary theme enhancement

Commit:
Translated using Weblate (Tajik)

Currently translated at 100.0% (521 of 521 strings)

Translation: UstadMobile/UI Strings
Translate-URL: http://weblate.ustadmobile.com/projects/ustad-mobile/ui-strings/tg/

mikedawson committed:
Merge remote-tracking branch 'origin/primary' into dev-crud-nav-login

kileha3 committed:
Animation: Removed obsolete animation related resources

mikedawson committed:
Merge pull request #153 from UstadMobile/primary-theme-enhancement

Removed obsolete animation related resources

mikedawson committed:
Update login endpoint

kileha3 committed:
Merge branch 'primary' of https://github.com/UstadMobile/UstadMobile into dev-crud-nav-login

varunasingh committed:
Tests espresos ok for clazzworkdetailoverview

kileha3 committed:
Added getstrarted and worspace link fragments

kileha3 committed:
Login: Added login fragment and it's presenter tests

kileha3 committed:
Login: Added worspace link presenter tests

kileha3 committed:
Login: Added getstarted espresso tests

kileha3 committed:
Implemented espresso tests

kileha3 committed:
Minor update on workspace link screen

kileha3 committed:
Disabled test cases

kileha3 committed:
Minor changes to avoid leakages

kileha3 committed:
Disabled test

kileha3 committed:
Enabled login espresso tests

kileha3 committed:
Minor update on presenter test: removing snackbar reference

kileha3 committed:
Minor updates on login presenters test

kileha3 committed:
Enabled all espresso tests

kileha3 committed:
Minor update on workspace link tests

kileha3 committed:
Changes as per pull request comments - WIP

kileha3 committed:
Updated login implementation

kileha3 committed:
Added link validator and formatter

kileha3 committed:
Added workspace args

kileha3 committed:
Updated workspace presenter and it's tests

kileha3 committed:
Minor changes on method signature

kileha3 committed:
Updated workspace implementation

kileha3 committed:
Added account list and presenters tests

kileha3 committed:
Finalized account list impl

kileha3 committed:
Minor change to support Dark mode

kileha3 committed:
Minor update on test cases

kileha3 committed:
Changed start destination

kileha3 committed:
Added accountlist espresso tests

mikedawson committed:
Add ProgressBar checking to CrudIdlingResource

Commit:
Merge remote-tracking branch 'origin/dev-crud-nav-login' into dev-crud-nav-login

Commit:
DI work in progress

kileha3 committed:
Minor update on fragments  onDestroy

mikedawson committed:
Fix CrudIdlingResource

Commit:
Merge branch 'dev-crud-nav-login' into dev-kodeindi

kileha3 committed:
Merge branch 'dev-crud-nav-login' of https://github.com/UstadMobile/UstadMobile into dev-crud-nav-login

mikedawson committed:
KodeIn DI work in progress

mikedawson committed:
Fix CrudIdlingResource to ignore hidden progress bars

kileha3 committed:
Merge branch 'dev-crud-nav-login' of https://github.com/UstadMobile/UstadMobile into dev-crud-nav-login

kileha3 committed:
Properly shutdown mockwebserver

Commit:
Merge branch 'dev-crud-nav-login' into dev-kodeindi

kileha3 committed:
Added jvm static on util functions

kileha3 committed:
Theory : Regular expression causing an issue - Disabled

mikedawson committed:
Fix use of Kodein-DI with jvmTestm

kileha3 committed:
Updated umaccount and reflect changes on account list

varunasingh committed:
Presenter and espresso test Ok

mikedawson committed:
Update Android fragments to compile using KodeIn-DI

kileha3 committed:
Tests: Updating workspace tests - WIP

kileha3 committed:
Workspace tests - WIP

mikedawson committed:
Remove obsolete TestApp.kt code
Fix ClazzLogListAttendanceFragmentTest
Fixes for UmAppDatabaseAndroidClientRule

Commit:
Merge branch 'dev-crud-nav-login' into dev-kodeindi

Commit:
Merge remote-tracking branch 'origin/dev-crud-nav-login' into dev-crud-nav-login

varunasingh committed:
Merge branch 'primary' into dev-crud-classwork

Added adb commands, fixed merge issues, test finish

kileha3 committed:
Workspace test impl -WIP

mikedawson committed:
Fix VideoContent to use KodeinDI

Commit:
done report end to end

Commit:
increase margins

varunasingh committed:
Test fix update 1

varunasingh committed:
Test fix2

varunasingh committed:
Test fix3

Commit:
webchunk converted to fragment

Commit:
webchunk espresso test

mikedawson committed:
Merge branch 'primary' into i18n-strings

Commit:
better name for webchunk test

mikedawson committed:
Bump version

mikedawson committed:
Update core to use KodeDI with a endpoint scope

varunasingh committed:
Getting PR ready

varunasingh committed:
PR update

mikedawson committed:
Update Android screens to use dependency injection

mikedawson committed:
Merge remote-tracking branch 'origin/primary' into primary

varunasingh committed:
Tests and wrap up . Commit before merge

kileha3 committed:
Updated workspace tests

kileha3 committed:
Added redirect fragment

kileha3 committed:
Added profile icon on host activity and created workspace route

varunasingh committed:
Merge branch 'primary' into dev-crud-classwork

mikedawson committed:
Adapt SharedSe network code to use dependency injection: work in progress

mikedawson committed:
Update DownloadDialogPresenterTest

kileha3 committed:
Remove account was sending read-only list which can't be updated - causing crashes

kileha3 committed:
Updated login cxml to avoid inflation exception

mikedawson committed:
Update clazz log creation to use dependency injection and operate on the correct database as per the endpoint

mikedawson committed:
Make app-ktor-server use dbcp2 with SQLite by default

mikedawson committed:
Set default container path if not included in JNDI

kileha3 committed:
Merge branch 'primary' of https://github.com/UstadMobile/UstadMobile into dev-crud-nav-login2

kileha3 committed:
Tested workspace link against server, working on login -WIP

varunasingh committed:
Fix presenter tests

kileha3 committed:
Updated login route

kileha3 committed:
Handled login and account list

mikedawson committed:
Work in progress to run download jvm test

kileha3 committed:
Updated tests

mikedawson committed:
Update ContainerEntryListResponder and tests to use dependency injection

mikedawson committed:
Add ConcatenatedContainerEntryFileResponderTest and update ConcatenatedContainerEntryFileResponder to use dependency injection

mikedawson committed:
Update ContainerEntryListResponder and tests to use dependency injection

mikedawson committed:
Update ContainerFetcherJobHttpUrlConnection and add basic test.

mikedawson committed:
DownloadJobItemRunner and test: Work-in-progress passing basic test cases

mikedawson committed:
Update ContainerDownloadManager and tests to use dependency injection

Commit:
Merge branch 'primary'

mikedawson committed:
Updates to Android to use dependency injection for network services and managers

Commit:
Merge branch 'primary'

Conflicts:
	app-android/src/main/java/com/ustadmobile/port/android/view/WebChunkActivity.kt
	app-android/src/main/res/navigation/mobile_navigation.xml

Commit:
updated with di

Commit:
test should fail now if screen didnt load after 10 seconds

varunasingh committed:
Merge branch 'primary' into dev-crud-classwork

mikedawson committed:
Update MountedContainerResponderTest

Commit:
wait for most of javascript to finish loading, test still doesnt pass on 21

Commit:
using right context

Commit:
assume true for tests and removed commented code

kileha3 committed:
Added missing logic when account is clicked from the account list

kileha3 committed:
Removed active account adapter and used only single adapter

kileha3 committed:
Resolved inflation issue on login layout - WIP

Commit:
changed context

mikedawson committed:
Merge pull request #155 from UstadMobile/dev-report-end

report end-to-end

Commit:
removed webchunk and video from old destmap

mikedawson committed:
Tidyup AccountListFragment

mikedawson committed:
Merge pull request #156 from UstadMobile/dev-crud-webchunk

Dev crud webchunk

kileha3 committed:
Update account list and about screen

varunasingh committed:
test passing clean up

mikedawson committed:
Fix UstadAccountManager to ensure that account state is saved to preferences and account last used time is recorded.

mikedawson committed:
Merge remote-tracking branch 'origin/primary' into primary

kileha3 committed:
Made bottom nav hidded when navigates away from main screens

kileha3 committed:
Merged primary

kileha3 committed:
Minor fixes after merging primary

kileha3 committed:
Added layout in nestedscrollview

varunasingh committed:
Presentertest fix

kileha3 committed:
Fixed minor logical error

kileha3 committed:
Minor changes and fixes

kileha3 committed:
Updated broken tests after changes on presenter

mikedawson committed:
Merge pull request #157 from UstadMobile/dev-crud-nav-login2

Dev crud nav login 2

kileha3 committed:
Minor changes of UX and UXI

kileha3 committed:
Reflected changes made on presenter to tests

kileha3 committed:
Minor test fixes

varunasingh committed:
Commit before merge

varunasingh committed:
Merge branch 'primary' into dev-crud-classwork

kileha3 committed:
Minor fixes on tests

kileha3 committed:
Updated tests to accomodate UXI change - showing snackbar on base fragment

kileha3 committed:
Minor test updates

mikedawson committed:
Work-in-progress integration of lib-http-testserver and androidclient test rule. Works to setup a test server and load data from the server.

mikedawson committed:
Merge pull request #160 from UstadMobile/dev-crud-nav-login3

Login UX improvements

mikedawson committed:
Merge remote-tracking branch 'origin/primary' into dev-networkdi

# Conflicts:
#	app-ktor-server/src/main/java/com/ustadmobile/lib/rest/UmRestApplication.kt

mikedawson committed:
Update UmRestApplication devMode handling of database for test containers

mikedawson committed:
Merge branch 'primary' of github.com:UstadMobile/UstadMobile into dev-networkdi

mikedawson committed:
Refactor redirect to move some logic to the presenter and add support for ARG_NEXT (e.g. for activity tests to automatically navigate to a preset destination)

mikedawson committed:
Misc updates to ensure that the basic download function works on emulator

varunasingh committed:
Integeration fixes.

mikedawson committed:
Remove deprecated UmAccountManager

mikedawson committed:
Update ContainerEntryListResponder and tests to use dependency injection

mikedawson committed:
Update ContentEntryOpener to use dependency injection

mikedawson committed:
Update LocalAvailabilityManager and associated classes

mikedawson committed:
Update LocalAvailabilityManagerImplTest

mikedawson committed:
Fix TestEmbeddedHTTPD

varunasingh committed:
Merge branch 'primary' into dev-crud-classwork

mikedawson committed:
Add ContainerDownloadManager function to add contents from a concatenatedinputstream

mikedawson committed:
Refactor ContentEntryType plugins and tests

mikedawson committed:
Fix BleEntryStatusTaskTest

mikedawson committed:
Add EntryStatusRequest class with test to wrap EntryStatusRequest contententryuids and endpoint info

varunasingh committed:
Espresso tests fix

varunasingh committed:
Yet another test fix.

mikedawson committed:
Fix BleGattSErveRcommon to use dependency injection and connect to the appropriate endpoint depending on the request received

mikedawson committed:
Remove obsolete BleHttpProxy and test (this was replaced by the BleHttpResponder

mikedawson committed:
Fix DownloadJobPreparerTest

mikedawson committed:
Remove unused import

mikedawson committed:
Disable NetworkManagerBleCommonTest for items that should be handled in ContainerDownloadManager

mikedawson committed:
Fix Gatt Server and NetworkManagerBle compilation on Android

mikedawson committed:
Disable ContentEntry2DetailFragmentTest download from server test that is not ready yet

mikedawson committed:
Ensure that NetworkManagerBle onCreate is called by DI module. Setup Repository online status to observe networkmanager connectivitystatus.

mikedawson committed:
Update multidexenabled flag on devMinApi21

mikedawson committed:
Refactor handling of entry status messages

varunasingh committed:
Some fixes to question assigning to ClazzWork

varunasingh committed:
Students view for ClassWork

mikedawson committed:
Commit connectivity status changes from networkmanager to the database automatically.

mikedawson committed:
Activate UmAppDatabaseSyncWorker

mikedawson committed:
Fix ContentEntryOpener

mikedawson committed:
Refactor BleMessage management: remove BleMessageTask and use sendMessage

mikedawson committed:
Update appconfig to avoid accidental sync with main db

varunasingh committed:
Merge branch 'primary' into dev-crud-classwork

mikedawson commented:
This pull request is now obsolete and replaced with #159
